gril/grilreply.c

```diff
@@ -493,23 +493,35 @@ struct reply_data_reg_state *g_ril_reply_parse_data_reg_state(GRil *gril,
 
 	g_ril_append_print_buf(gril, "{");
 
+	if (g_ril_vendor(gril) == OFONO_RIL_VENDOR_SAMSUNG_MSM_8226) {
+		/* DANGER WILL ROBINSON
+         * In some cases from Vodaphone we are receiving a RAT of 102
+         * while in tunnels of the metro. Let's assume that if we
+         * receive 102 we actually want a RAT of 2 for EDGE service */
+/* 		if (str_arr->num_str > 4 && strcmp(str_arr->str[0], "1") && strcmp(str_arr->str[3], "102")) {
```

Comment:
Not sure if we need that: It sounds more like a fix for Vodafone (and its not clear in which countries). I can include this if someone tells me how to replace the string in the struct - probably copying everything into a new one?